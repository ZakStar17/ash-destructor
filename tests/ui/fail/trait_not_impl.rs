use ash_destructor::DeviceDestroyable;

#[path = "../../utils/mod.rs"]
mod utils;

use utils::ImplDeviceDestroyable;

#[derive(DeviceDestroyable)]
struct Named {
    a: ImplDeviceDestroyable,
    // doesn't implement trait
    b: String,
}

#[derive(DeviceDestroyable)]
struct Unnamed(ImplDeviceDestroyable, String);

fn main() {}
