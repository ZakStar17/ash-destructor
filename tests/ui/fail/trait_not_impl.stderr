error[E0277]: the trait bound `String: DeviceDestroyable` is not satisfied
  --> tests/ui/fail/trait_not_impl.rs:12:5
   |
12 |     b: String,
   |     ^^^^^^^^^ the trait `DeviceDestroyable` is not implemented for `String`
   |
   = help: the following other types implement trait `DeviceDestroyable`:
             &T
             Box<T>
             Buffer
             BufferView
             CommandPool
             DescriptorPool
             DescriptorSetLayout
             DescriptorUpdateTemplate
           and $N others

error[E0277]: the trait bound `String: DeviceDestroyable` is not satisfied
  --> tests/ui/fail/trait_not_impl.rs:15:10
   |
15 | #[derive(DeviceDestroyable)]
   |          ^^^^^^^^^^^^^^^^^ the trait `DeviceDestroyable` is not implemented for `String`
16 | struct Unnamed(ImplDeviceDestroyable, String);
   |                                       ------ required by a bound introduced by this call
   |
   = help: the following other types implement trait `DeviceDestroyable`:
             &T
             Box<T>
             Buffer
             BufferView
             CommandPool
             DescriptorPool
             DescriptorSetLayout
             DescriptorUpdateTemplate
           and $N others
